### ISSUE TITLE
How to fix the warning warn("Output seems independent of input.")

### ISSUE THREAD

**Issue by G1iderK** [XanaduAI/pennylane#589]
When I run a quantum neural network, a warning ("Output seems independent of input.") has happened. However, when I check my code it has no bug in it. Here is my train process 
```
for i in range(n_epochs):
        params = opt.step(lambda par:cost(params,x_train,labels), params)
        entropys=cost(params,x_train,labels)
``` 
And my loss function in define as 
```
def cost(params,input,labels):
    outputs = [circuit(params, x) for x in input]
    entropys=0
    for i in range(100):
        output=outputs[i]
        label=labels[i]
        entropys=entropys+cross_entropy(output,label)
    return entropys
```
.When I run the 
```params = opt.step(lambda par:cost(params,x_train,labels), params)```
the params are not updated.  And the warning
```
 UserWarning: Output seems independent of input.
  warnings.warn("Output seems independent of input.")
```
has happened.
The structure of my parameters is
`params=2*np.pi*np.random.randn(n_layers, 12)`
Has anybody met this problem?

**Comment by josh146** [comment]
Hi @G1iderK! The warning `Output seems independent of input.` comes from Autograd, and means that Autograd is unable to determine how the output of your cost function (`entropys`) depends on the input `params`.

Could you post what your `cross_entropy` and `circuit` functions look like?

Note that we also have a [PennyLane discussion forum](https://discuss.pennylane.ai), which might be more suited to debugging help!

**Comment by G1iderK** [comment]
@josh146 Thx josh146, here is my cross_entropy code
```
def cross_entropy(output, label):
    entropy = 0
    for k in range(4):
        entropy = entropy - label[k] * np.log(output[k])
    return entropy
```
and my `def circuit()`
```
@qml.qnode(dev)
def circuit(params, input=None):
    n_layers = 5

    n=np.zeros(4)
    n[3]=1
    temp=np.kron(input,n)
    qml.templates.AmplitudeEmbedding(features=temp, wires=range(4))

    for i in range(n_layers):
        for k in range(4):
            qml.RZ(params[i][k], wires=k)
            qml.RX(params[i][k + 4], wires=k)
            qml.RZ(params[i][k + 8], wires=k)

        qml.CNOT(wires=[1, 2])
        qml.CNOT(wires=[0, 1])
        qml.CNOT(wires=[2, 3])

    # Measurement producing 4 classical output values
    measure_list = [2, 3]

    return qml.probs(wires=measure_list)
```
And I have already fix this problem just now. I found that I write `circuit(params, input)`  before and now I rewrite it as `circuit(params, input=None)`. It seems like the parameter `input` is understood as a variable parameter without predefine and it can not be calculated the gradient. Thx all the same.: ）

**Comment by josh146** [comment]
Hi @G1iderK, yes that's correct --- non-trainable QNode arguments are specified as keyword arguments 🙂 Glad it's now working!
